source/torque_vector/main.c

```diff
@@ -36,39 +34,41 @@ GPIOInitConfig_t gpio_config[] = {
     GPIO_INIT_OUTPUT(SPI_CS_MAG_GPIO_Port, SPI_CS_MAG_Pin, GPIO_OUTPUT_HIGH_SPEED),
 
     // GPS USART
-    GPIO_INIT_USART3RX_PC5,
-    GPIO_INIT_USART3TX_PC4,
+    GPIO_INIT_UART4RX_PA1,
+    GPIO_INIT_UART4TX_PA0,
+
+    // GPS Auxillary pins
+    GPIO_INIT_OUTPUT(GPS_RESET_GPIO_Port, GPS_RESET_Pin, GPIO_OUTPUT_LOW_SPEED),
 
     // EEPROM
     GPIO_INIT_OUTPUT(NAV_EEPROM_CS_GPIO_PORT, NAV_EEPROM_CS_PIN, GPIO_OUTPUT_HIGH_SPEED),
     GPIO_INIT_OUTPUT(NAV_WP_GPIO_PORT, NAV_WP_PIN, GPIO_OUTPUT_HIGH_SPEED),
 
     // CAN
     GPIO_INIT_CANRX_PA11,
-    GPIO_INIT_CANTX_PA12};
+    GPIO_INIT_CANTX_PA12
+    };
 
-/* USART Configuration */
-// M9N GPS
-dma_init_t usart_gps_tx_dma_config = USART3_TXDMA_CONT_CONFIG(NULL, 1);
-dma_init_t usart_gps_rx_dma_config = USART3_RXDMA_CONT_CONFIG(NULL, 2);
-usart_init_t huart_gps = {
+// GPS USART Configuration
+dma_init_t usart_gps_tx_dma_config = USART4_TXDMA_CONT_CONFIG(NULL, 1);
+dma_init_t usart_gps_rx_dma_config = USART4_RXDMA_CONT_CONFIG(NULL, 2);
+usart_init_t huart_gps =
+{
     .baud_rate = 115200,
     .word_length = WORD_8,
     .hw_flow_ctl = HW_DISABLE,
-    .mode = MODE_TX_RX,
     .stop_bits = SB_ONE,
     .parity = PT_NONE,
     .obsample = OB_DISABLE,
     .ovsample = OV_16,
-    .adv_feature.rx_inv = false,
-    .adv_feature.tx_inv = false,
-    .adv_feature.auto_baud = false,
-    .adv_feature.data_inv = false,
-    .adv_feature.msb_first = false,
-    .adv_feature.overrun = false,
-    .adv_feature.dma_on_rx_err = false,
+    .periph      = UART4,
+    .wake_addr = false,
+    .usart_active_num = USART4_ACTIVE_IDX,
+    .tx_errors = 0,
+    .rx_errors = 0,
     .tx_dma_cfg = &usart_gps_tx_dma_config,
-    .rx_dma_cfg = &usart_gps_rx_dma_config};
+    .rx_dma_cfg = &usart_gps_rx_dma_config
+};
 
 #define TargetCoreClockrateHz 16000000
```

Comment:
will eventually have to check that bumping up the clock rate doesn't break anything